### DISCUSSION TITLE
Clarification Needed on .gitignore Rules for the broadcast Directory

### DISCUSSION THREAD

**Discussion by moeed7** [Cyfrin/foundry-full-course-cu#885] [Q&A]
Hello everyone,

I've been reviewing the .gitignore file and noticed a section that I'd like to get some clarification on. Specifically, the rules concerning the broadcast/ directory:

,,,
#!/broadcast
/broadcast/*/31337/
/broadcast/**/dry-run/
/broadcast
,,,

From my understanding:

We're specifying not to ignore the broadcast directory with !/broadcast.
We're ignoring specific subdirectories within broadcast/ using patterns.
Finally, we're ignoring the entire broadcast/ directory again.
This seems a bit redundant or conflicting. Could someone help clarify the intent behind these rules? Are all these patterns necessary, or can we simplify this section for better clarity?

Thank you for the insights! (And yes, GPT4 organized the question for me)

**Comment by shrutic11** [comment]
When we do git init, it automatically creates a .gitignore file with certain folders, and patterns added to the list.

I don't think it comes with #!/broadcast because anything followed by a # is a comment and only for readability purposes.

the # infront of !/broadcast is commenting the statement, so you are NOT not to ignore the broadcast directory with !/broadcast. (or simply not caring as to what that command does.)

Let's say it was !/broadcast initially -> So, yes, it doesn't want to ignore the entire broadcast folder
but indeed the next two sub folders based on patterns 
!/broadcast
/broadcast/*/31337/
/broadcast/**/dry-run/

The reason till this statement could be to upload rest of the broadcast folders, except the executions for dry-run and 31337 (Anvil).

I think the /broadcast is manually added by Patrick, because I heard him say in one of the videos that he doesn't like to push the broadcast folder.
